kolibri/core/content/views.py

```diff
@@ -290,7 +290,11 @@ def get(self, request, zipped_filename, embedded_filepath):
         zipped_path = get_path_or_404(zipped_filename)
 
         # Sometimes due to URL concatenation, we get URLs with double-slashes in them, like //path/to/file.html.
-        # Normalize the path by converting those double-slashes to a single slash.
+        # If it is a leading double slash (i.e. /zipcontent/filename.zip//file.html) then it will get passed
+        # to zipcontent as a single leading slash - detect this and remove if present.
+        if embedded_filepath.startswith("/"):
+            embedded_filepath = embedded_filepath[1:]
```

Comment:
The comment assumes knowledge of the url pattern defined externally in `urls.py`, but by the time it gets here that's already been parsed out. Perhaps:

```suggestion
        # If the embedded file path has a leading double slash (i.e. //file.html) it should get passed
        # to zipcontent with a single leading slash instead.
        if embedded_filepath.startswith("/"):
            embedded_filepath = embedded_filepath[1:]
```

The method signature could also be elaborated upon. For example:

```python
    def get(self, request, zipped_filename, embedded_filepath):
        """
        Handles GET requests and serves a static file from within the zip file.

        zipped_filename is the name of a zip file, and embedded_filepath is a path to something within that zip file.
        """
```

Reply:
I'm also surprised based on the comment that the code isn't this instead:

```python
        if embedded_filepath.startswith("//"):  # double slash
            embedded_filepath = embedded_filepath[1:]
```

Reply:
`embedded_filepath` will not start with '//', because we tell Django that the URL is formatted as '/zipcontent/<zipped_filename>/<embedded_filepath>`. Notice the `/` in between `<zipped_filename>` and `<embedded_filepath>`. So if the URL is:

`/zipcontent/1234.zip//filename.html`

then when `get` is called, `zipped_filename = "1234.zip"` and `embedded_filepath = "/filename.html"` (not `//filename.html`). 

 As `embedded_filepath` should be a relative path to the file, and `/filename.html` is an absolute path starting from the root of a filesystem, we want to strip that `/` from the URL.

Reply:
ah OK - thanks for clarifying. So there is no leading double slash! In that case, I would write:

```suggestion
        # If the embedded file path has a leading slash (e.g. /file.html), it should get passed
        # to zipcontent with no leading slash.
        if embedded_filepath.startswith("/"):
            embedded_filepath = embedded_filepath[1:]
```